components/director/internal/authenticator/middleware.go

```diff
@@ -49,17 +49,18 @@ func (a *Authenticator) Handler() func(next http.Handler) http.Handler {
 			}
 
 			claims := Claims{}
-
-			token, err := jwt.ParseWithClaims(bearerToken, &claims, a.getKeyFunc())
+			_, err = jwt.ParseWithClaims(bearerToken, &claims, a.getKeyFunc())
 			if err != nil {
 				wrappedErr := errors.Wrap(err, "while parsing token")
 				log.Error(wrappedErr)
-				http.Error(w, wrappedErr.Error(), http.StatusUnauthorized)
-				return
-			}
 
-			if !token.Valid {
-				http.Error(w, "Invalid token", http.StatusUnauthorized)
+				vErr, ok := err.(*jwt.ValidationError)
```

Comment:
It just can be handled by the `isInvalidTenantError` func.

Reply:
Nope, it can't, as it is custom error type and I want to compare types, not error messages. I'm using `isInvalidTenantError` line below, on inner validation error.

Reply:
Maybe then you can wrap this `vErr, ok := err.(*jwt.ValidationError)` into func isJWTValidationError?

Reply:
Hmm... It will be hard to do so, as I need inner error of the `Validation Error`, so the function would also return two arguments (like `convertToValidationError`). I think there is no benefit to do so.